### DISCUSSION TITLE
Calculating average euler angles for individual grains

### DISCUSSION THREAD

**Discussion by fboschetty** [mtex-toolbox/mtex#1858] [Ask Anything]
Hi,

I have a series of EBSD maps of single grains. There is some noise in the maps and I would like to calculate an average set of euler angles for each map. Ideally I would like to calculate a weighted average and weighted standard deviation, where the the weights are in the inverse of the MAD for each pixel. Is there a straightforward way to do this in MTEX?

Thanks,

Felix

**Comment by kilir** [comment]
Hi,

orientation/mean does take weights e.g.
```matlab
mtexdata titanium
% compute grains and ebsd with grainID
[grains,ebsd.grainId]=ebsd.calcGrains;

% looping is slow but easier to do the weights
for i=1:length(grains)
mO(i) = mean(ebsd(grains(i)).orientations,'weights',ebsd(grains(i)).ci);
end
```
Cheers,
Rüdiger

**Reply by fboschetty** [reply]
Hi Rüdiger,

That's great, thank you. Is there a similar way to calculate the variance or standard deviation?

Thanks,

Felix
